src/main/java/relations/r1cs/FileToR1CS.java

```diff
@@ -0,0 +1,478 @@
+        // Assumes input to be ordered by row and that the last line is blank.
+        final LinearCombination<FieldT> L = new LinearCombination<>();
+
+        try {
+            String nextLine;
+            reader.mark(1);  // TODO - this is very strange.
+            while ((nextLine = reader.readLine()) != null) {
+                String[] tokens = nextLine.split(" ");
+
+                int col = Integer.parseInt(tokens[0]);
+                int row = Integer.parseInt(tokens[1]);
+                final BN254aFields.BN254aFr value = new BN254aFields.BN254aFr(tokens[2]);
+
+                if (row == index) {
+                    L.add(new LinearTerm<>(col, (FieldT) value));
+                } else if (row > index) {
+                    reader.reset();
+                    return new Tuple2<>(L, reader);
+                }
+            }
+        } catch (Exception e){
+            System.err.println("Error: " + e.getMessage());
+        }
+        return new Tuple2<>(L, reader);
+    }
+
+    private static <FieldT extends AbstractFieldElementExpanded<FieldT>>
+        JavaPairRDD<Long, LinearTerm<FieldT>>
+    makeDistributedCombinationFromJSON(
+            Configuration config,
+            ArrayList<Integer> partitions,
+            JSONArray[] constraintArray,
+            int constraintArrayIndex,
+            int numPartitions,
+            int numConstraints
+    ){
+        return config.sparkContext()
+                .parallelize(partitions, numPartitions).flatMapToPair(part -> {
+            final long partSize = part == numPartitions ?
+                    numConstraints % (numConstraints / numPartitions) : numConstraints / numPartitions;
+
+            final ArrayList<Tuple2<Long, LinearTerm<FieldT>>> T = new ArrayList<>();
+            for (long i = 0; i < partSize; i++) {
+                final long index = part * (numConstraints / numPartitions) + i;
+
+                JSONObject next = (JSONObject) constraintArray[(int) index].get(constraintArrayIndex);
+                Iterator<String> keys = next.keySet().iterator();
+                while (keys.hasNext()) {
+                    String key = keys.next();
+
+                    long columnIndex = Long.parseLong(key);
+                    BN254aFields.BN254aFr value;
+                    try{
+                        value = new BN254aFields.BN254aFr((String) next.get(key));
+                    } catch (ClassCastException e){
+                        // Handle case when key-value pairs are String: Long
+                        value = new BN254aFields.BN254aFr(Long.toString((long) next.get(key)));
+                    }
+                    T.add(new Tuple2<>(index, new LinearTerm<>(columnIndex, (FieldT) value)));
+                }
+
+            }
+            return T.iterator();
+        });
+    }
+
+    private static <FieldT extends AbstractFieldElementExpanded<FieldT>>
+        JavaPairRDD<Long, LinearTerm<FieldT>>
+    makeDistributedCombinationFromStream(
```

Comment:
Is this used anywhere?

Reply:
Yes. It is used in a place that was previously commented out. This comment is outdated since my changes this morning anyway. It also has a better name now.